x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/action/TransportPutRollupJobAction.java

```diff
@@ -115,6 +121,15 @@ public void onFailure(Exception e) {
         });
     }
 
+    static void checkForDeprecatedTZ(PutRollupJobAction.Request request) {
+        String timeZone = request.getConfig().getGroupConfig().getDateHistogram().getTimeZone();
+        String modernTZ = DateUtils.DEPRECATED_LONG_TIMEZONES.get(timeZone);
```

Comment:
DateUtils.of can also do that for you, but you would get a different log message.. Do you think it is worth using instead?

Reply:
also possibly similar logging/validation should be in RangeQueryBuilder, but there is a separate PR for this https://github.com/elastic/elasticsearch/issues/38449

Reply:
Will take a look at DateUtils.of, thanks!  I'm not too picky about the error message contents, as long as it describes what's going on.  Would rather reuse code where possible :)

Reply:
Had a look, `DateUtils.of()` only checks the existing "short" names, while this check is for the "long" names that the IANA have deprecated.  We can modify `DateUtils.of()` to do this check too... but I'll do that in a followup so that this PR doesn't balloon with extra non-rollup deprecations.